docs/incidents/2024-08-12-site-outage.md

```diff
@@ -0,0 +1,72 @@
+# Site Outage
+
+## What happened?
+
+- 2023-06-16
+  - Dependency update to `CRI-O v1.27` in production.
+- 2024-08-??
+  - AppArmor problems observed in production system logs: access denied during shutdown of containers.
+  - A workflow that involves manual cleanup of containers during deployment is introduced as a workaround.
+  - Deployments continue to take place.
+  - System log error messages begin to accumulate rapidly -- some of these are the access-denied messages, and some relate to missing resources that have been manually removed.
+- 2024-08-10
+  - 13:05 Disk space in production is exhausted due to the container teardown/resource reclamation error messages.
+  - 13:05 Service outage occurs.
+- 2024-08-12
+  - 15:45 - The site outage is noticed by operational staff.
+  - 15:51 - A reboot is initiated.
+  - 16:00 - Disk space is identified as the problem; space is reclaimed and service restoration begins.
+  - 16:02 - System recovery is hampered by the fact that redeployments cause further error logging.
+  - 16:04 - Given the identification of container log message accumulation as a probable cause of the outage, a decision is made to begin opportunistic upgrades in the hope that these will resolve the problem.
+  - 16:08 - Journal/log files are removed to restore disk space.
+  - 16:09 - Service recovers, albeit with continuing error messages.
+  - 16:12 - Operating system update to `Ubuntu 24.04` begins.
+  - 16:58 - Operating system update completed.
+  - 16:43 - Container redeployment begins.
+  - 17:08 - Container runtime: updated to `CRI-O v1.30`
```

Comment:
I don't think it has been confirmed yet why this upgrade appears to have solved the problem.

Launchpad bug https://bugs.launchpad.net/apparmor/+bug/2067900 referenced from [this comment](https://github.com/openculinary/infrastructure/pull/43#discussion_r1735026047) has not yet reached Ubuntu 24.04 -- perhaps the containers were unexpectedly temporarily running in `unconfined` mode (this appears to contribute to the problem), or a change in a recent version of CRI-O removed the preconditions for the problem?

Reply:
I think that https://github.com/containerd/containerd/pull/10123 was the fix; updating our CRI-O dependencies will have updated our `runc`, and as a result that resolved the permissions problem.

I believe that we were running Ubuntu 23.10 prior to the upgrade -- so the AppArmor v4.0.0 rules that caused the problem would indeed have been in place.

We should add this to the root cause notes, although it may not affect the (already in-progress) suggested action items.